solutions/cf-1713E.mdx

```diff
@@ -0,0 +1,145 @@
+---
+id: cf-1713E
+source: Codeforces Div 2
+title: Cross Swapping
+author: Aakash Gokhale
+---
+
+## Solution
+
+[Official Analysis](https://codeforces.com/blog/entry/105419)
+
+Notice that $a[i][j]$ can only be swapped with $a[j][i]$ no matter which operations are performed.
+
+In addition, we want to lexicographically sort the array. Thus, we can iterate and greedily process the array.
+
+For all $j$ such that $i < j$ (we only need to consider each pair once and keeping $i < j$ will ensure this), we can determine whether we want to swap them or keep them the same.
+
+If $a[i][j] > a[j][i]$, we want to swap them. If $a[i][j] < a[j][i]$ we want to keep them the same. If they are equal it doesn't matter.
+
+However, it is possible that not all conditions can work together with each other and swapping something means we can't swap something else later on.
+
+We can use the [Amogus Trick](https://codeforces.com/blog/entry/106407) to represent this problem. Create a disjoint set union of $2n$ nodes where node $i$ represents the state where the operation is performed on $i$ and $i + n$ represents the state where operation is not performed on $i$.
```

Comment:
is there no other way to solve this problem?
because if the only way to solve this is with some meme-a** trick, i don't think this is a good problem

Reply:
or unless you can give a nice intuitive method for arriving at this trick

Reply:
i wouldn't mention the trick then if it isn't really needed

Reply:
fixed